src/api/trackerEvents.ts

```diff
@@ -0,0 +1,165 @@
+import { D2ApiGeneric } from "./d2Api";
+import { Id, Selector, D2ApiResponse } from "./base";
+import { Preset, FieldPresets, D2Geometry } from "../schemas";
+import { getFieldsAsString } from "./common";
+import _ from "lodash";
+
+export class TrackerEvents {
+    constructor(public api: D2ApiGeneric) {}
+
+    get<Fields extends D2TrackerEventFields>(
+        params: EventsParams<Fields>
+    ): D2ApiResponse<TrackerEventsResponse> {
+        return this.api.get<TrackerEventsResponse>("/tracker/events", {
+            ..._.omit(params, ["fields"]),
+            fields: getFieldsAsString(params.fields),
+        });
+    }
+}
+
+type ProgramStatus = "ACTIVE" | "COMPLETED" | "CANCELLED";
+type IsoDate = string;
+type Username = string;
+type CommaDelimitedListOfUid = string;
+type CommaDelimitedListOfAttributeFilter = string;
+type CommaDelimitedListOfDataElementFilter = string;
+type UserInfo = {
+    uid: Id;
+    username: string;
+    firstName: string;
+    surname: string;
+};
+type EventStatus = "ACTIVE" | "COMPLETED" | "VISITED" | "SCHEDULED" | "OVERDUE" | "SKIPPED";
+type IdScheme = string;
+
+interface D2TrackerEventBase {
+    event: Id;
+    status: EventStatus;
+    program: Id;
+    programStage?: Id;
+    enrollment?: Id;
+    enrollmentStatus?: "ACTIVE" | "COMPLETED" | "CANCELLED";
+    orgUnit: Id;
+    orgUnitName?: string;
+    relationships?: [];
+    occurredAt: IsoDate;
+    scheduledAt?: IsoDate;
+    storedBy?: Username;
+    followup?: boolean;
+    deleted?: boolean;
+    createdAt?: IsoDate;
+    updatedAt?: IsoDate;
+    createdBy?: UserInfo;
+    attributeOptionCombo?: Id;
+    attributeCategoryOptions?: Id;
+    updatedBy?: UserInfo;
+    dataValues: DataValue[];
+    notes?: string[];
```

Comment:
Here I'm not sure if notes is a `string[]` now in the new API. Checking the documentation I've found this https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/tracker.html#tracker-notes but instead of this, maybe for the moment I would leave it as an array as @9sneha-n has done in `D2TrackerEnrollment`

Reply:
Yes, it seems it's now an object, not a plain string. When in doubt, create objects and check the real response of the API.

Reply:
Got it! Fixed it to just an array.